cmd/data-usage.go

```diff
@@ -177,10 +181,92 @@ func loadDataUsageFromBackend(ctx context.Context, objAPI ObjectLayer) (DataUsag
 type Item struct {
 	Path string
 	Typ  os.FileMode
+
+	bucket     string // Bucket.
+	prefix     string // Only the prefix if any, does not have final object name.
+	objectName string // Only the object name without prefixes.
+	lifeCycle  *lifecycle.Lifecycle
```

Comment:
can we not fetch the lifecycle as necessary from the `globalBucketMetadataSys()` ? rather than keeping a copy here.

Reply:
That would just be extra overhead.

Reply:
why the value is cached in memory - that is the source of truth, keeping a copy would mean that you are not applying the latest rules IMO there is no reason to cache it here.

Reply:
It is, there will be millions of requests and having to look that up in a map will be a significant penalty.

If this was a user triggered action it should be, since there would be inconsistencies, but it is by definition out of the users hands. Added lifecycles will be picked up on the next crawler cycle anyway.

Performance is critical for the base crawler.